corehq/elastic.py

```diff
@@ -23,27 +23,62 @@
 from corehq.pillows.mappings.xform_mapping import XFORM_INDEX_INFO
 
 
+def _es_hosts():
+    es_hosts = getattr(settings, 'ELASTICSEARCH_HOSTS', None)
+    if not es_hosts:
+        es_hosts = [settings.ELASTICSEARCH_HOST]
+
+    hosts = [
+        {
+            'host': host,
+            'port': settings.ELASTICSEARCH_PORT,
+        }
+        for host in es_hosts
+    ]
+    return hosts
+
+
 def get_es_new():
     """
     Get a handle to the configured elastic search DB.
     Returns an elasticsearch.Elasticsearch instance.
     """
     if not getattr(get_es_new, '_es_client', None):
-        es_hosts = getattr(settings, 'ELASTICSEARCH_HOSTS', None)
-        if not es_hosts:
-            es_hosts = [settings.ELASTICSEARCH_HOST]
-
-        hosts = [
-            {
-                'host': host,
-                'port': settings.ELASTICSEARCH_PORT,
-            }
-            for host in es_hosts
-        ]
+        hosts = _es_hosts()
         get_es_new._es_client = Elasticsearch(hosts)
     return get_es_new._es_client
 
 
+def get_es_export():
+    """
+    Get a handle to the configured elastic search DB with settings geared towards exports.
+    Returns an elasticsearch.Elasticsearch instance.
+    """
+    if not getattr(get_es_export, '_es_client', None):
+        hosts = _es_hosts()
+        get_es_export._es_client = Elasticsearch(
+            hosts,
+            retry_on_timeout=True,
+            max_retries=3,
+            # Timeout in seconds for an elasticsearch query
+            timeout=30,
+        )
+    return get_es_export._es_client
+
+ES_DEFAULT_INSTANCE = 'default'
+ES_EXPORT_INSTANCE = 'export'
+
+ES_INSTANCES = {
+    ES_DEFAULT_INSTANCE: get_es_new,
+    ES_EXPORT_INSTANCE: get_es_export,
+}
+
+
+def get_es_instance(es_instance=ES_DEFAULT_INSTANCE):
```

Comment:
When I first read this PR I thought that `es_instance` referred to a completely different es cluster, but it looks like it's actually a different `Elasticsearch` object that points to the same cluster, but with different settings. I wonder if it would be more clear to future readers of the code if these kwargs were called `es_client` instead? (I don't feel strongly about this name change but just thought I would throw it out there in case you agree it's worth changing.)